File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.java
class Solution {
    public int maxLength(List<String> arr) {
        List<String> r=new ArrayList<String>();
        r.add("");
        for(String str : arr)
        {
            if(!isUnique(str))
                continue;
            List<String> s=new ArrayList<String>();
            for(String n : r)
            {
                String temp=n+str;
                if(isUnique(temp))
                    s.add(temp); 
            }
            r.addAll(s);
        }
        int max=0;
        for(String a: r)
            max=Math.max(max,a.length());
        return max;
    }
    public boolean isUnique(String str)
    {
        if(str.length() > 26)
            return false;
        boolean[] map= new boolean[26];
        for(char c : str.toCharArray())
        {
            if(map[c-'a'])
                return false;
            else
                map[c-'a']=true;
        }
        return true;
    }
}

File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.py
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        d = [set()]
        for a in arr:
            if len(set(a)) < len(a):
                continue;
            a=set(a)
            for c in d[:]:
                if a & c:
                    continue;
                d.append(a|c)
        b=0
        for p in d:
            if b < len(p):
                b=len(p)
        return b